plugins/glustershd/actor.go

```diff
@@ -2,49 +2,66 @@ package glustershd
 
 import (
 	"errors"
+	"path"
 
 	"github.com/gluster/glusterd2/glusterd2/daemon"
 	"github.com/gluster/glusterd2/glusterd2/volume"
 	"github.com/gluster/glusterd2/glusterd2/xlator"
 	gderrors "github.com/gluster/glusterd2/pkg/errors"
 
 	log "github.com/sirupsen/logrus"
+	config "github.com/spf13/viper"
 )
 
 var names = [...]string{"replicate", "afr"}
 
 const (
-	selfHealKey = "self-heal-daemon"
+	selfHealKey          = "self-heal-daemon"
+	granularEntryHealKey = "granular-entry-heal"
 )
 
 type shdActor struct{}
 
 func (actor *shdActor) Do(v *volume.Volinfo, key string, value string, logger log.FieldLogger) error {
-	if key != selfHealKey {
+	if key != selfHealKey && key != granularEntryHealKey {
 		return nil
 	}
-	glustershDaemon, err := newGlustershd()
-	if err != nil {
-		return err
-	}
+	switch key {
+	case selfHealKey:
+		glustershDaemon, err := newGlustershd()
+		if err != nil {
+			return err
+		}
 
-	switch value {
-	case "on":
-		if isHealEnabled(v) {
-			err = daemon.Start(glustershDaemon, true, logger)
-			if err != nil && err != gderrors.ErrProcessAlreadyRunning {
-				return err
+		switch value {
+		case "on":
+			if isHealEnabled(v) {
+				err = daemon.Start(glustershDaemon, true, logger)
+				if err != nil && err != gderrors.ErrProcessAlreadyRunning {
+					return err
+				}
+			}
+		case "off":
+			if !isHealEnabled(v) {
+				isVolRunning, err := volume.AreReplicateVolumesRunning()
+				if err != nil {
+					return err
+				}
+				if !isVolRunning {
+					return daemon.Stop(glustershDaemon, true, logger)
+				}
 			}
 		}
-	case "off":
-		if !isHealEnabled(v) {
-			isVolRunning, err := volume.AreReplicateVolumesRunning()
+	case granularEntryHealKey:
+		switch value {
+		case "enable":
+			var options []string
+			glusterdSockpath := path.Join(config.GetString("rundir"), "glusterd2.socket")
+			options = append(options, "granular-entry-heal-op", "glusterd-sock", glusterdSockpath)
```

Comment:
why do we need `append` over here?
can this be done as `options:=[]string{"granular-entry-heal-op", "glusterd-sock", glusterdSockpath}`